plugins/selectionhighlight.lua

```diff
@@ -56,3 +56,43 @@ function DocView:draw_line_body(line, x, y)
   return line_height
 end
 
+-- Special thanks to Adam(aethy)
+-- for writing code which served as a guide
+
+--- This added feature displays selection highlights in/on the scrollbar
+
+local old_draw_docview_scrollbar = DocView.draw_scrollbar
+
+function DocView:draw_scrollbar()
+  old_draw_docview_scrollbar(self)
+  local scrollbar = self.v_scrollbar
+
+  local mw = 5  --marker width
+  local mh = 8  -- marker height
+  local cmh = 4  --current selection marker height
+  local cm_color =  {common.color "#DA70D6"} --current selection marker color
+  
+  local line1, col1, line2, col2 = get_selection(self.doc, true)
+  
+  if line1 == line2 and col1 ~= col2 then
```

Comment:
So, this will only be drawn if we're selecting exactly one line? It won't show anything for multiline selections? Is that correct?

Reply:
I guess this matches the behaviour above, so that's OK.